lib/dataloader.ex

```diff
@@ -156,12 +161,32 @@ defmodule Dataloader do
         end)
         |> Map.new()
 
-      %{dataloader | sources: sources}
+      updated_dataloader = %{dataloader | sources: sources}
+
+      emit_stop_event(id, start_time_mono, updated_dataloader)
+
+      updated_dataloader
     else
       dataloader
     end
   end
 
+  defp emit_start_event(id, system_time, dataloader) do
+    :telemetry.execute(
+      [:dataloader, :batches, :run, :start],
```

Comment:
Let's call these `[:dataloader, :source, :run, :start]` since what we're timing here is the execution of a particular source.